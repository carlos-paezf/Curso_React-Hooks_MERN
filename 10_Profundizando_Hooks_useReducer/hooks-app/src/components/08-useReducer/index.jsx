export { TodoApp } from "./TodoApp";
export { TodoList } from "./TodoList";
export { TodoListItem } from "./TodoListItem";
export { TodoAdd } from "./TodoAdd";