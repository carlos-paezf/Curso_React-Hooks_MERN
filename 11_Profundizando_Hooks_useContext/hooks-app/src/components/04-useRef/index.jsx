export { FocusScreen } from './FocusScreen'
export { RealExampleRef } from './RealExampleRef'