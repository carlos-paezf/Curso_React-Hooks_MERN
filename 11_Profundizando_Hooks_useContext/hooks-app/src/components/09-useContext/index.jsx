export { AppRouter } from "./AppRouter";

export { AboutScreen } from "./AboutScreen";
export { HomeScreen } from "./HomeScreen";
export { LoginScreen } from "./LoginScreen";
export { MainApp } from "./MainApp";

export { NavBar } from "./NavBar";