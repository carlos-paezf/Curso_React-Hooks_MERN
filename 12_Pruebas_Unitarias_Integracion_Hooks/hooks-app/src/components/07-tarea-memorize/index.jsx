export { Hijo } from "./Hijo";
export { Padre } from "./Padre";