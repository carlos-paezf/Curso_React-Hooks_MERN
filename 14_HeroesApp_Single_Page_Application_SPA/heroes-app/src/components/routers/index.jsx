export { AppRouter } from './AppRouter'
export { DashboardRoutes } from './DashboardRoutes'