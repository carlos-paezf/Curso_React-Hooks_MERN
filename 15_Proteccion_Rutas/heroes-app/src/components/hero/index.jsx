export { HeroScreen } from './HeroScreen'
export { HeroList } from './HeroList'
export { HeroCard } from './HeroCard'