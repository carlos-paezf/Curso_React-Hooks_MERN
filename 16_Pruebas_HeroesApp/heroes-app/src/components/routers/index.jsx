export { AppRouter } from './AppRouter'
export { DashboardRoutes } from './DashboardRoutes'
export { PublicRoute } from './PublicRoute'
export { PrivateRoute } from './PrivateRoute'