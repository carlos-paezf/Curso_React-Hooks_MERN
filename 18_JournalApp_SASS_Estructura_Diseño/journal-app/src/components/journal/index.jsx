export { JournalEntries } from "./JournalEntries";
export { JournalEntry } from "./JournalEntry";
export { JournalScreen } from "./JournalScreen";
export { NothingSelected } from "./NothingSelected";
export { Sidebar } from "./Sidebar";