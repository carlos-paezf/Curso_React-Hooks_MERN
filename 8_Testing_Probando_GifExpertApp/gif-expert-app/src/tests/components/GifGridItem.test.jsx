import { shallow } from 'enzyme'
import { GifGridItem } from '../../components';


describe('Pruebas en <GifGridItem />', () => {
    const title = 'Gif para los test'
    const url = 'http://localhost:1234/img.gif'

    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de tener un párrafo con el título', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    });

    test('Debe de tener una imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        expect(img.props().src).toBe(url)
        expect(img.props().alt).toBe(title)
    });

    test('Debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div')
        expect(div.props().className.includes('animate__fadeIn')).toBe(true)
    })
})
