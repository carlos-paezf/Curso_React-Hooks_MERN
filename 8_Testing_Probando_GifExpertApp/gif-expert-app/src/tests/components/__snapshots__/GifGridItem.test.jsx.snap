// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Gif para los test"
    className="card animate__animated animate__fadeInDown"
    src="http://localhost:1234/img.gif"
  />
  <p>
    Gif para los test
  </p>
</div>
`;
